x/archivedb/key.go

```diff
@@ -0,0 +1,103 @@
+// Copyright (C) 2019-2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package archivedb
+
+import (
+	"bytes"
+	"encoding/binary"
+	"errors"
+
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/utils/wrappers"
+)
+
+var (
+	ErrInsufficientLength = errors.New("packer has insufficient length for input")
+	keyHeight             = []byte{1}
+)
+
+const (
+	internalKeySuffixLen = wrappers.LongLen
+	minInternalKeyLen    = internalKeySuffixLen + 1
+)
+
+// dbKey
+//
+// The keys contains a few extra information alongside the given key. The key is
+// what is known outside of this internal scope as the key, but this struct has
+// more information compacted inside the dbKey to take advantage of natural
+// sorting. The inversed height is stored right after the Prefix, which is
+// MaxUint64 minus the desired Height.
+//
+// The inverse height is stored, instead of the height, as part of the key. That
+// is because the database interface only provides ascending iterators.
+// Archivedb's main usage is to query a given key at a given height, or the
+// immediate previous height. That is why the heights are being converted from
+// `height` to `MAX_INT64 - height`.
+//
+//	Example (Asumming MAX_INT64 is 10,000):
+//	 |  User given  |  Stored as   |
+//	 |--------------|--------------|
+//	 |    foo:10    |   foo:9910   |
```

Comment:
Stored as values don't appear to be `MAX_INT64 - height`. Would be 9990 and 9980, no?